dot/network/block_announce.go

```diff
@@ -145,7 +145,7 @@ func (s *Service) getBlockAnnounceHandshake() (Handshake, error) {
 
 	return &BlockAnnounceHandshake{
 		Roles:           s.cfg.Roles,
-		BestBlockNumber: uint32(latestBlock.Number),
+		BestBlockNumber: uint32(latestBlock.Number), //nolint:gosec
```

Comment:
Just curious to know why we need this tag here.
What does it do?

Reply:
At least some of those casts could lead to integer overflow. I think we should at some point go through all of them, check and potentially address them instead of disabling the lint globally.

Reply:
Yea unfortunately we do a lot of casting like this since we have to conform to the types in specific message types.  It would probably be best to do a slight refactor to avoid all these type casts in the first place and use the right type initially.